src/request/containers/RequestDetailPanelDataOperationCommandContainer.ts

```diff
@@ -0,0 +1,33 @@
+import { getSelectedOperation } from '../selectors/RequestDetailDataSelectors';
+import { IRequestState } from '../stores/IRequestState';
+import { RequestDetailPanelDataOperationCommand, IRequestDetailPanelDataOperationCommandProps } from '../components/RequestDetailPanelDataOperationCommand';
+
+import { connect } from 'react-redux';
+
+function getLanguageForDatabase(database: string): string {
+    switch (database) {
+        case 'MongoDB':
+            return 'json';
+        default: 
+            return 'sql';
+    }
+}
+
+function mapStateToProps(state: IRequestState, ownProps): IRequestDetailPanelDataOperationCommandProps {
+    const operation = getSelectedOperation(state);
+
+    return {
+        command: operation ? operation.command : '',
+        language: getLanguageForDatabase(operation ? operation.database : undefined)
+    }    
+}
+
+function mapDispatchToProps(dispatch) {
+    return {
+    };
+}
+
+export const RequestDetailPanelDataOperationCommandContainer = connect(
+    mapStateToProps,
+    mapDispatchToProps
+)(RequestDetailPanelDataOperationCommand);
```

Comment:
Yes, did read the post (and similar others).  There seems to be a fairly liberal application of the smart vs. dumb component idea--liberal in the sense of a nearly 1:1 mapping between them as opposed to trying to minimize the number of smart components.  The posts suggest that the need for dumb components to pass down properties to their child components, without using them itself, is often a sign that there should be a smart component in between.

As the Data tab evolves it's going to have more and more sub-components (as each sub-tab of the detail panel are added), which would compound the problem of passing down the set of properties necessary for each subcomponent.  If each of those has their own, corresponding smart component, then it becomes very obvious what is responsible for getting/setting their properties.